lib/mnesia/src/mnesia_dumper.erl

```diff
@@ -404,8 +404,12 @@ dets_insert(Op,Tab,Key,Val, Storage0) ->
 	    dets_updated(Tab,Key),
 	    mnesia_lib:db_match_erase(Storage, Tab, Val);
 	clear_table ->
-	    dets_cleared(Tab),
-	    ok = mnesia_lib:db_match_erase(Storage, Tab, '_')
+            %% Val is a match_delete pattern
+            case Val of
+                '_' -> dets_cleared(Tab);
+                _ -> dets_updated(Tab, Val)
```

Comment:
These `dets_cleared/1`, `dets_incr_counter/2`, `dets_updated/2` look rather unclear. They store updates tree in a proc dict: https://github.com/emqx/otp/blob/f899f860e2f8ff93bc5f1af3e6802a1d58ad1d5c/lib/mnesia/src/mnesia_dumper.erl#L415
The tree is eventually erased and abandoned: https://github.com/emqx/otp/blob/f899f860e2f8ff93bc5f1af3e6802a1d58ad1d5c/lib/mnesia/src/mnesia_dumper.erl#L204
Anyway, calling `dets_updated(Tab, Val)` seems safe, calling `dets_updated(Tab, Key)` makes no sense, since Key is `'_'`: https://github.com/emqx/otp/blob/f899f860e2f8ff93bc5f1af3e6802a1d58ad1d5c/lib/mnesia/src/mnesia.erl#L2838